internal/rbd/manager.go

```diff
@@ -268,3 +316,108 @@ func (mgr *rbdManager) CreateVolumeGroup(ctx context.Context, name string) (type
 
 	return vg, nil
 }
+
+func (mgr *rbdManager) CreateVolumeGroupSnapshot(
+	ctx context.Context,
+	vg types.VolumeGroup,
+	name string,
+) (types.VolumeGroupSnapshot, error) {
+	// TODO: reserve a unique name for the snapshot
+	pool, err := vg.GetPool(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// volumeNamePrefix is an optional parameter, can be an empty string
+	prefix := mgr.parameters["volumeNamePrefix"]
+
+	// journalPool is an optional parameter, use pool if it is not set
+	journalPool, ok := mgr.parameters["journalPool"]
+	if !ok || journalPool == "" {
+		journalPool = pool
+	}
+
+	clusterID, err := vg.GetClusterID(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cluster id for volume group snapshot %q: %w", vg, err)
+	}
+
+	uuid, freeUUID, err := mgr.getGroupUUID(ctx, clusterID, journalPool, name, prefix)
+	defer func() {
+		// no error, no need to undo the reservation
+		if err == nil {
+			return
+		}
+
+		freeUUID()
+	}()
+
+	monitors, err := util.Mons(util.CsiConfigFile, clusterID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find MONs for cluster %q: %w", clusterID, err)
+	}
+
+	_ /*journalPoolID*/, poolID, err := util.GetPoolIDs(ctx, monitors, journalPool, pool, mgr.creds)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate a unique CSI volume group with uuid for %q: %w", uuid, err)
```

Comment:
cleanup  
```suggestion
		return nil, fmt.Errorf("failed to get PoolID for JournalPool %s and ImagePool %s: %w", journalPool, pool, err)
```
or in simple just return the err as it is.

Reply:
Thanks, will address this too.